chain/src/chain.rs

```diff
@@ -882,6 +892,26 @@ impl Chain {
 		// Validate the full kernel history (kernel MMR root for every block header).
 		self.validate_kernel_history(&header, &txhashset)?;
 
+		// sandbox validation ok, go to overwrite txhashset and clean the sandbox
+		{
+			// Before overwriting, drop file handles in underlying txhashset
+			{
+				let mut txhashset_ref = self.txhashset.write();
+				txhashset_ref.release_backend_files();
+			}
+
+			txhashset::zip_write(self.db_root.clone(), txhashset_data, &header, false)?;
```

Comment:
1. We extract the zip file out into the sandbox.
2. Then validate the contents of the sandbox.
3. Then if valid we (re)extract the zip file out into the real location.

(3) Feels wrong to me. I thinkwe should take the _validated_ contents of the sandbox and copy them across to the txhashset dir. We should not (re)extract the zipfile.

The 2nd extract step just increases the possibility of us doing it _slightly_ differently the 2nd time, particularly with us passing in a different flag (true/false) the 2nd time.

Proposal - 

1. Extraction into sandbox
2. Validate sandbox
3. Copy across validated files from sandbox into txhashset dir

Reply:
Accept.
Change from (re)extract to folder move (rename).